src/logDownloaderService.py

```diff
@@ -0,0 +1,79 @@
+import os
+import time
+import hashlib
+import subprocess
+import psutil
+from git import Repo
+import shutil
+from loguru import logger
+
+LOGS_DIRECTORY = os.getenv('LOGS_DIRECTORY', '/path')
+ALL_LOGS_FLAG = os.getenv('ALL_LOGS', 'false').lower() == 'true'
+GIT_REPO_PATH = '/path'
+REMOTE_REPO = 'origin'
+BRANCH = 'main'
+ADVANTAGE_SCOPE_PATH = '/path'
+
+
+def hash_file(filepath):
+    hasher = hashlib.md5()
+    with open(filepath, 'rb') as f:
+        buf = f.read()
+        hasher.update(buf)
+    return hasher.hexdigest()
```

Comment:
Just a basic example of `pathlib` usage:
```suggestion
def get_file_hash(filepath: Path) -> str:
    content = filepath.read_bytes()
    return hashlib.md5(content).hexdigest()
```
1. If the function returns something, the name should be clear.
2. Please use type hints throughout the module.
3. You can use `hashlib` functions directly on the input.